app-creator/src/widgets/draggable-widget/draggable-widget.ts

```diff
@@ -36,18 +58,125 @@ export class DraggableWidget extends LitElement {
+      editable,
+      hasOverlay,
+      handleDragstart,
+      handleDragend,
+      handleRemoveWidget,
+    } = this;
 
     const overlay = hasOverlay ? html`<div class="overlay"></div>` : nothing;
 
+    const editableMarkup = editable
+      ? html`
+          <div id="editable-view">
+            <iron-icon class="edit-buttons" icon="create"></iron-icon>
+            <iron-icon
+              class="edit-buttons"
+              icon="icons:delete"
+              @click=${handleRemoveWidget}
+            ></iron-icon>
+          </div>
+        `
+      : nothing;
+
     return html`
-      <div id="container" draggable="true">
+      <div
+        id="container"
+        draggable="true"
+        @dragstart=${handleDragstart}
+        @dragend=${handleDragend}
+      >
         <slot></slot>
-        ${overlay}
+        ${overlay} ${editableMarkup}
       </div>
     `;
   }
+
+  /**
+   * Triggered when the trash icon is clicked. If the widget is the last in the dropzone,
+   * we display the empty placeholder and center the container's flex alignments.
+   */
+  handleRemoveWidget() {
+    const parent = this.parentElement;
+    if (parent == null) {
+      return;
+    }
+
+    parent.removeChild(this);
+
+    const childrenCount = parent.childElementCount;
+
+    // We never really remove the placeholder div (we just hide it with display='none').
+    // When the children count is 1 after removing a widget, we want to unhide the placeholder.
+    if (childrenCount === 1) {
+      this.showPlaceholder(parent);
+    }
+  }
+
+  /**
+   * Callback triggered on the beginning of a drag action. We use it to reference the currently
+   * dragged element in a global state.
+   * @param e dragstart event
+   */
+  handleDragstart(e: Event) {
+    const target = e.target as HTMLDivElement;
+    if (target == null) {
+      return;
+    }
+
+    // We want to unwrap the draggable wrapper and only reference the the inner element.
+    const widget = target?.querySelector('slot')?.assignedElements()[0];
+    if (widget == null) {
+      return;
+    }
+
+    // Referencing the currently dragged element in global state.
+    store.draggingElement = widget;
+  }
+
+  /**
+   * Callback triggered on the end of a drag action. We use it to clear the reference
+   * of a currently dragged element and increment a widget id if necessary.
+   * @param e dragend event
+   */
+  handleDragend() {
+    const draggingElement = store.draggingElement;
+    if (draggingElement == null) {
+      store.resetDraggingValues();
+      return;
+    }
+
+    if (store.elementAdded && !store.reordering) {
+      store.widgetIDs[draggingElement.id]++;
+    }
+
+    store.resetDraggingValues();
+  }
+
+  /**
+   * Displays placeholder by changing display property from 'none' to 'flex'.
+   * @param parent: element containing placeholder content
+   */
+  showPlaceholder(parent: HTMLElement) {
+    const placeholder = parent.querySelector(
+      '#empty-placeholder'
```

Comment:
I noticed that empty-placeholder is used in a few places. Could we make it a variable here and import it in dropzone-widget.ts?
`export const PLACEHOLDER_ID = "empty-placeholder"`